packages-next/fields-document/src/relationship-data.tsx

```diff
@@ -41,16 +41,27 @@ export function addRelationshipData(
       }
       return [];
     } else {
+      // Inline
       const id = data?.id;
       if (id != null) {
         const labelField = getLabelFieldsForLists(graphQLAPI.schema)[relationship.listKey];
-        let val = await graphQLAPI.run({
-          query: `query($id: ID!) {item:${
-            relationship.listKey
-          }(where: {id:$id}) {${labelFieldAlias}:${labelField}\n${relationship.selection || ''}}}`,
-          variables: { id },
-        });
-
+        let val;
+        try {
+          val = await graphQLAPI.run({
+            query: `query($id: ID!) {item:${
+              relationship.listKey
+            }(where: {id:$id}) {${labelFieldAlias}:${labelField}\n${
+              relationship.selection || ''
+            }}}`,
+            variables: { id },
+          });
+        } catch (err) {
+          // If we're unable to find the user (e.g. we have a dangling reference), or access was denied
+          // then simply return null.
+          console.error(`Unable to fetch relationship data: relationship: ${JSON.stringify(relationship)}, id: ${id} `);
+          console.error(err);
```

Comment:
I'm undecided as to whether these console error messages are helpful, or just noise 🤔

Reply:
Yeah, i don't think they're helpful though could you look at what it looks like to only do this on access control/does not exist? (the easiest way might be to do a many query and just get the first one from the array and handle if the array is empty?)